### DISCUSSION TITLE
Unshipped orders report with prices?

### DISCUSSION THREAD

**Discussion by c-gross** [amzn/selling-partner-api-models#3656] [General]
Hi,

I'm about implement the new SP-API since we are still using the old MWS API.
We need all orders, also including the customization zip url, since we mainly sell personalized products.

As far as I can see, the order API is still pointless, like it was in MWS as well, since it has the same low request limitations so I have to go again the way of using reports to be able to process all orders. (Whats the point of the order API, just to trick new dev into a dead end?)

To get new orders I was about to take the _GET_FLAT_FILE_ACTIONABLE_ORDER_DATA_SHIPPING_. That contains the customization zip url but it's missing the paid product prices. Since there are no example report CSVs to see what we get I need to ask, if there is a better report type that will provide new/unshipped orders with all informations to ship them and the customization zip url including paid product prices.

**Comment by cml43140** [comment]
Hey you ever have any luck finding prices?  The getOrders API is pointless due to limitations, and looking to go this report route to get all unshipped orders.  Prices are critical for us too though, any advice appreciated!!

**Reply by c-gross** [reply]
I'm using GET_ORDER_REPORT_DATA_SHIPPING report. It contains the prices but is missing taxes.

**Reply by cml43140** [reply]
Thanks for the reply!  Does that one similarly to the other one, also drop off the report once the order is fulfilled?  Also, curious what duration you programmed in to "wait" after calling the generate report API, until you go and retrieve it?  I've used 60 seconds which seems to work so far, but am still in building phase, and was curious if you found a threshold that consistently works?  Thanks for any advice!

**Reply by c-gross** [reply]
I don't remember if they get removed from the report bug I think so. I created a temp table to store all orders for the import and ignore already imported orders. Beside that I run the reports with timerange of a few hours anyway.

I run the report request after 2min since last year xmas was showing 1min can be too short. But I also have cronjobs to retry up to a specific amount after that, if the report was not ready until the next report gets requested.

It's not perfect but it's working so far.

**Reply by cml43140** [reply]
Perfect, was thinking along the same lines with temp table in case they don't fall off, etc.  2 mins sounds good as well, thanks again for the replies
